src/libraries/System.Reflection.DispatchProxy/src/System/Reflection/DispatchProxy.cs

```diff
@@ -11,6 +11,7 @@ namespace System.Reflection
     /// </summary>
     public abstract class DispatchProxy
     {
+        private static readonly Type s_dispatchProxyType = typeof(DispatchProxy);
```

Comment:
Caching `typeof` like this is unnecessary. It is actually going to be de-optimization with recent JIT changes.